include/execution.hpp

```diff
@@ -172,32 +174,27 @@ namespace std::execution {
       sender<S> &&
       __has_sender_types<sender_traits<remove_cvref_t<S>>>;
 
+  // NOT TO SPEC:
+  template <typed_sender Sender,
+            template <class...> class Tuple,
+            template <class...> class Variant>
+    using value_types_of_t =
+      typename sender_traits<decay_t<Sender>>::template
+        value_types<Tuple, Variant>;
+
+  // NOT TO SPEC:
+  template <typed_sender Sender, template <class...> class Variant>
+    using error_types_of_t =
+      typename sender_traits<decay_t<Sender>>::template error_types<Variant>;
+
   template<class S, class... Ts>
     concept sender_of =
       typed_sender<S> &&
-      same_as<
-        __types<Ts...>,
-        typename sender_traits<S>::template
-            value_types<__types, type_identity_t>>;
-
-  template <class... As>
-    requires (sizeof...(As) != 0)
-    struct __front;
-  template <class A, class... As>
-    struct __front<A, As...> {
-      using type = A;
-    };
-  template <class... As>
-    requires (sizeof...(As) == 1)
-    using __single_t = __t<__front<As...>>;
-  template <class... As>
-    requires (sizeof...(As) <= 1)
-    using __single_or_void_t = __t<__front<As..., void>>;
+      same_as<__types<Ts...>, value_types_of_t<S, __types, __single_t>>;
 
   template<class S>
     using __single_sender_value_t =
-      typename sender_traits<remove_cvref_t<S>>
-        ::template value_types<__single_or_void_t, __single_or_void_t>;
+      value_types_of_t<S, __single_or_void_t, __single_t>;
```

Comment:
I am a bit unsure why we need the duplication of `__single_or_void_t` and `__single_t` Shouldnt the latter be subsumed by the former?